src/main/java/ru/ayaz/User.java

```diff
@@ -0,0 +1,63 @@
+package ru.ayaz;
+
+import java.util.Set;
+import java.util.TreeSet;
+
+public class User {
+    private String nickname;
+    private Set<String> ignoredUsersSet;
+
+
+    public User() {
+        this.ignoredUsersSet = new TreeSet<String>();
+    }
+
+    public void setNickname(String nickname) throws invalidNicknameException {
+        validateNickName(nickname);
+        this.nickname = nickname;
+    }
+
+    private void validateNickName(String nickname) throws invalidNicknameException {
+        if (nickname == null) {
+            throw new invalidNicknameException(nickname);
+        }
+
+        if (isSpacesOnly(nickname)) {
+            throw new invalidNicknameException(nickname);
+        }
+
+        if (isFirstCharDigit(nickname)) {
+            throw new invalidNicknameException(nickname);
+        }
+    }
+
+    private boolean isFirstCharDigit(String nickname) {
+        return Character.isDigit(nickname.charAt(0));
+    }
+
+    private boolean isSpacesOnly(String nickname) {
+        return nickname.replaceAll("\\s+", "").equals("");
+    }
+
+
+    public String getNickname() {
+        return nickname;
+    }
+
+    public boolean isInIgnoredSet(String nickname) {
+        return ignoredUsersSet.contains(nickname);
+    }
+
+    public boolean ignoreUser(String blackNickname) {
```

Comment:
&#x1F34E; **Issue**
It is not clear, that this method returns anything just from looking at its name. The impression is that user with a given name will be ignored no matter what.
To fix it you have two options:
1) rename method to, for example, "tryIgnoringUser" which will give a hint that something might go wrong and we need to check for that (which you don't do by the way in your ChatServer class when executing a user command, meaning if the user wants to ignore himself - he can't and he doesn't get any response if he tries to ignore himself, the method just returns "false" which is not checked anywhere. It might be a wanted behavior, but defenitely something to think about.)
2) you can return "void" and throw checked exception when you can't ignore given user. Checked exceptions are another way to convey the outcome of an operation.